SauceLabs.Visual/VisualApi.cs

```diff
@@ -1,3 +1,97 @@
-﻿namespace SauceLabs.Visual;
+﻿using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Reflection;
+using System.Threading.Tasks;
+using GraphQL;
+using GraphQL.Client.Http;
+using GraphQL.Client.Serializer.Newtonsoft;
+using Newtonsoft.Json;
+using OpenQA.Selenium;
+using SauceLabs.Visual.GraphQL;
+using SauceLabs.Visual.Utils;
 
-public class VisualApi { }
+namespace SauceLabs.Visual
+{
+    internal class VisualApi<T> where T : IHasCapabilities, IHasSessionId
+    {
+        private readonly string _username;
+        private readonly string _accessKey;
+        private readonly GraphQLHttpClient _graphQlClient;
+
+        public VisualApi(T webdriver, DataCenter dc, string username, string accessKey, HttpClient? httpClient = null)
+        {
+            _username = username.Trim();
+            _accessKey = accessKey.Trim();
+
+            if (_username == "" || _accessKey == "")
+            {
+                throw new VisualClientException(
+                    "Invalid SauceLabs credentials. Please check your SauceLabs username and access key at https://app.saucelabs.com/user-setting");
+            }
+
```

Comment:
my recommendation is to use `string.IsNullOrEmpty`
```suggestion
            if (string.IsNullOrWhiteSpace(username) || string.IsNullOrWhiteSpace(accessKey)) {
            {
                throw new VisualClientException(
                    "Invalid SauceLabs credentials. Please check your SauceLabs username and access key at https://app.saucelabs.com/user-setting");
            }
             _username = username.Trim();
            _accessKey = accessKey.Trim();

```